src/components/CharacterGallery.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { getCharacters } from '../util/swapi';
+import CharacterItem from './CharacterItem';
+import './CharacterGallery.css';
+
+class CharacterGallery extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            characters: [],
+            selectedGender: 'all'
+        };
+    }
+
+    async componentDidMount() {
+        const characters = await getCharacters(this.props.urls);
+        this.setState({ characters: characters });
+    }
+
+    showMaleGender = () => {
+        this.setState({ selectedGender: 'male' });
+    }
+
+    showFemaleGender = () => {
+        this.setState({ selectedGender: 'female' });
+    }
+
+    showAllGenders = () => {
+        this.setState({ selectedGender: 'all' });
+    }
+
+    makeCharacterItems() {
+        const characterItems = [];
+
+        for (let i = 0; i < this.state.characters.length; i++) {
```

Comment:
Instead of rendering the results of this method, the better solution is to use an array map inside the render method.

And even more an array map is preferred over declaring an empty array then for loop and push to it.

Reply:
As a general practice, if you need a separate function to render JSX, you actually want a separate component for that. In this case tho, just mapping over characters and returning a <CharacterItem /> is enough.

Reply:
replaced `makeCharacterItems` with `map()` in CharacterGallery.js